### PR TITLE
Expand apollo-cache code coverage

### PR DESCRIPTION
**Pull request by jamesmillerburgess**
First time contributor here!

I've added some tests to one of the more sparse files while familiarizing myself with the codebase.

### CODE REVIEW

File: packages/apollo-cache/src/__tests__/cache.ts [hunk 1]
```diff
@@ -1,13 +1,114 @@
...
+    it('either writes a fragment or a query', () => {
+      const test = new TestCache();
+      test.read = jest.fn();
+      test.writeFragment = jest.fn();
+      test.writeQuery = jest.fn();
+
+      test.writeData({});
+      expect(test.writeQuery).toBeCalled();
+
+      test.writeData({ id: 1 });
```
Comment (jamesmillerburgess) [inline packages/apollo-cache/src/__tests__/cache.ts]: There is a truthy check on `id` in `writeData`, hence we can't set `id` to `0` in the test. Should this be using something stricter, like `typeof id !== 'undefined'` or `hasOwnProperty`?

Comment (jbaxleyiii) [reply packages/apollo-cache/src/__tests__/cache.ts]: @jamesmillerburgess yeah this an edge case I don't think we have run into yet, but a move to typeof would probably be better!

**Comment by jbaxleyiii** [comment]
@jamesmillerburgess this is AMAZING! Thank you so much for the help here! 🎉 I've left one comment but this is looking excellent!

**Comment by jamesmillerburgess** [comment]
@jbaxleyiii This should be good to go now!

**Pull request merged**